filecoin-proofs/src/api/post.rs

```diff
@@ -187,7 +191,48 @@ pub fn generate_post(
 
     let unique_trees_res: Vec<_> = unique_challenged_replicas
         .into_par_iter()
-        .map(|(id, replica)| replica.merkle_tree(sector_size).map(|tree| (id, tree)))
+        .map(|(id, replica)| {
+            // TODO: The cache-root should be provided by the caller, but for now we'll
+            // just create a directory in a known location
+            let mut cache = create_dumb_cache(id).unwrap();
+
+            let CachedTreeAbsPaths { top_half, leaves } = cache.paths(MerkleTreeCacheKey::CommR());
+
+            println!("checking merkle tree (leaves) cache-file: {:?}", leaves);
+
+            let tree: std::result::Result<_, failure::Error> = if Path::new(&leaves).exists() {
+                println!("merkle tree (leaves) cache-file existed: {:?}", leaves);
+
+                let mut leaves_cache_file = File::open(leaves.clone()).unwrap();
+
+                leaves_cache_file.seek(SeekFrom::Start(0)).unwrap();
+
+                let mut top_half_cache_file = OpenOptions::new()
+                    .create(true)
+                    .write(true)
+                    .truncate(false)
+                    .read(true)
+                    .open(top_half.clone())
+                    .unwrap();
+
+                top_half_cache_file.seek(SeekFrom::Start(0)).unwrap();
+
+                MerkleTree::new_with_files(
```

Comment:
@schomatis - This call is producing an `Err` in my branch. I suspect it has something to do with the permission bit used when I open the file(s).